libs/milvus/langchain_milvus/vectorstores/zilliz.py

```diff
@@ -71,53 +72,74 @@ class Zilliz(Milvus):
-
-                try:
-                    self.col.create_index(
-                        self._vector_field,
-                        index_params=self.index_params,
-                        using=self.alias,
-                    )
-
-                # If default did not work, most likely Milvus self-hosted
-                except MilvusException:
-                    # Use HNSW based index
-                    self.index_params = {
-                        "metric_type": "L2",
-                        "index_type": "HNSW",
-                        "params": {"M": 8, "efConstruction": 64},
-                    }
-                    self.col.create_index(
-                        self._vector_field,
-                        index_params=self.index_params,
-                        using=self.alias,
-                    )
-                logger.debug(
-                    "Successfully created an index on collection: %s",
-                    self.collection_name,
-                )
-
-            except MilvusException as e:
-                logger.error(
-                    "Failed to create an index on collection: %s", self.collection_name
-                )
-                raise e
+            embeddings_functions: List[EmbeddingType] = self._as_list(
+                self.embedding_func
+            )
+            vector_fields: List[str] = self._as_list(self._vector_field)
+            if self.index_params is None:
+                indexes_params: List[dict] = [
+                    {} for _ in range(len(embeddings_functions))
+                ]
+            else:
+                indexes_params = self._as_list(self.index_params)
+
+            for i, embeddings_func in enumerate(embeddings_functions):
+                if not self._get_index(vector_fields[i]):
+                    try:
+                        # If no index params, use a default HNSW based one
+                        if not indexes_params[i]:
+                            if self._is_sparse_embedding(embeddings_func):
+                                indexes_params[i] = {
+                                    "metric_type": "IP",
+                                    "index_type": "SPARSE_INVERTED_INDEX",
+                                    "params": {"drop_ratio_build": 0.2},
+                                }
+                            else:
+                                indexes_params[i] = {
+                                    "metric_type": "L2",
+                                    "index_type": "AUTOINDEX",
+                                    "params": {},
+                                }
+
+                        try:
+                            self.col.create_index(
+                                vector_fields[i],
+                                index_params=indexes_params[i],
+                                using=self.alias,
+                            )
+
+                        # If default did not work, most likely on Zilliz Cloud
```

Comment:
ditto

Reply:
Should I keep "most likely on Zilliz Cloud"?  maybe:
`If default did not work (most likely on Zilliz Cloud), use *HNSW* index instead`
and then I can also remove the next comment (line 119)